internal/controlplane/kuadrant_control_plane_controller.go

```
@@ -0,0 +1,265 @@
+		readyCondition.Reason = kuadrantv1beta1.ControlPlaneReasonComponentsHealthy
+	}
+
+	meta.SetStatusCondition(&cp.Status.Conditions, readyCondition)
+
+	if err := r.Status().Update(ctx, cp); err != nil {
+		r.logger.Error(err, "failed to update status")
+	}
+}
+
+func (r *ControlPlaneReconciler) buildComponentStatuses(ctx context.Context) []kuadrantv1beta1.ComponentStatus {
+	var statuses []kuadrantv1beta1.ComponentStatus
```

golangci-lint (2.12.2)

[error] Consider preallocating statuses with capacity len(r.deployer.EnabledComponents())

(prealloc)